src/main/java/com/alvarium/hash/Sha256Provider.java

```diff
@@ -0,0 +1,26 @@
+package com.alvarium.hash;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+import com.alvarium.utils.Encoder;
+
+public class Sha256Provider implements HashProvider {
+
+  private final MessageDigest sha256;
+
+  public Sha256Provider() throws HashTypeException {
```

Comment:
Make this (and the other provider constructors) `protected` so that classes outside of this package can't arbitrarily instantiate the classes directly.

Reply:
fixed the sha256 provider constructor and class visibility in 6f0daf5c304725c358a1a53af7551c23296baab5 and added further documentation.